Build failure: No rule to make target libg2o.so

The ROS2 build sometimes fails due to a race condition, e.g., from https://github.com/clydemcqueen/orca3/issues/6:

~~~
make[2]: *** No rule to make target '/root/orca_ws/src/orb_slam_2_ros/orb_slam2/Thirdparty/g2o/lib/libg2o.so', needed by '/root/orca_ws/src/orb_slam_2_ros/orb_slam2/lib/liborb_slam2_ros_core.so'.  Stop.
~~~

The fix is to depend on the CMake targets `DBoW2` and `g2o`, not the `.so` files. I will submit a PR.

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -101,8 +101,8 @@ ament_target_dependencies(orb_slam2_ros_core
 target_link_libraries(orb_slam2_ros_core
   ${OpenCV_LIBS}
   ${EIGEN3_LIBS}
-  ${PROJECT_SOURCE_DIR}/orb_slam2/Thirdparty/DBoW2/lib/libDBoW2.so
-  ${PROJECT_SOURCE_DIR}/orb_slam2/Thirdparty/g2o/lib/libg2o.so
+  DBoW2
+  g2o
 )
 
 # map serialization addition - library boost serialization 

